### PR TITLE
docs: optimize site layout

### PR DESCRIPTION
**Pull request by MadCcc**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 💡 Background and solution
1. 调整了 5.0 新组件在侧边栏的位置，并添加了 New 标签
2. 给快乐主题加了小红点
3. header 设置为了 sticky，固定 layout，侧边栏可以一直滚到底部查看全部组件

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     -      |
| 🇨🇳 Chinese |     -      |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 24d702c</samp>

This pull request introduces a new theme editor feature for the ant-design library. It adds a new theme token `contentMarginTop` and uses it to adjust the layout and alignment of the site content. It also refactors some components to use theme token values and utility functions for styling and rendering. It adds a badge to the theme switch button to highlight the new feature. It also adds a `tag` property to the frontmatter of some component documentation files to mark them as new features.

### <samp>🤖 Generated by Copilot at 24d702c</samp>

*  Add `tag` property to the frontmatter of the documentation files for the new components `app`, `float-button`, `qr-code`, and `tour` to indicate that they are new features ([link](https://github.com/ant-design/ant-design/pull/44111/files?diff=unified&w=0#diff-c779f7d3b6a66acd0ce5e8bf4f1625badb983056711cc82435808368588ed38cR9), [link](https://github.com/ant-design/ant-design/pull/44111/files?diff=unified&w=0#diff-88905ff2c3941f956e0d1dbf8871b8f2d633992532d79997fc5be05032729291R10), [link](https://github.com/ant-design/ant-design/pull/44111/files?diff=unified&w=0#diff-80a107045d1a20a6bb129a0a9728f40fbdb0844205cc57fb4ea23271b99e8adcR9), [link](https://github.com/ant-design/ant-design/pull/44111/files?diff=unified&w=0#diff-778117125c1b0d160ec5e0ff086b12c8118fdad8b97d28f351948704a1fe1e11R10), [link](https://github.com/ant-design/ant-design/pull/44111/files?diff=unified&w=0#diff-97715d670d42f6224c59e289b477e433969f9948894a96f9e078c0b0d2139335R11), [link](https://github.com/ant-design/ant-design/pull/44111/files?diff=unified&w=0#diff-80e8f648f6b8a80820c718716bff7aa341374a962e9ecd4bbe2242e4f7ec1e59R12), [link](https://github.com/ant-design/ant-design/pull/44111/files?diff=unified&w=0#diff-1ad8b229c044ca1a06a7bbed19bc03af75d71e159a8b70caf2a52aa2e328b38aR9), [link](https://github.com/ant-design/ant-design/pull/44111/files?diff=unified&w=0#diff-8eeca01fdc999bcb32068854b6930fb0df7f5374284c2d03e3a0b7fb6d553d8cR10))
*  Move the `float-button` component from the `other` group to the `general` group in the documentation files ([link](https://github.com/ant-design/ant-design/pull/44111/files?diff=unified&w=0#diff-80a107045d1a20a6bb129a0a9728f40fbdb0844205cc57fb4ea23271b99e8adcL3-R3), [link](https://github.com/ant-design/ant-design/pull/44111/files?diff=unified&w=0#diff-778117125c1b0d160ec5e0ff086b12c8118fdad8b97d28f351948704a1fe1e11L3-R3))
*  Add a new component `ComponentItem` to render each component card in the components list page and extract it from the `ComponentsList` component ([link](https://github.com/ant-design/ant-design/pull/44111/files?diff=unified&w=0#diff-12f648feaf3caa0382edb3b014fdd7dbedcad48bee322535d7cd01c5d210c9e6R93-R134), [link](https://github.com/ant-design/ant-design/pull/44111/files?diff=unified&w=0#diff-12f648feaf3caa0382edb3b014fdd7dbedcad48bee322535d7cd01c5d210c9e6L236-L273))
*  Modify the `offsetTop` prop of the `Affix` component in the `ComponentsList` and `IconSearch` components to ensure that the search bar is affixed below the header when scrolling ([link](https://github.com/ant-design/ant-design/pull/44111/files?diff=unified&w=0#diff-f1df5cf9560d26c9862a4824e7d1a13f6a8bec1ec7338e9327f3e7a6a74c1585L142-R142), [link](https://github.com/ant-design/ant-design/pull/44111/files?diff=unified&w=0#diff-8b799f7fc1319c44839ab462e8211e9e598d446618f5770aac912faebc536077L127-R127))
*  Add a `badge` prop with a `dot` value to the `FloatButton.Group` component in the `ThemeSwitch` component to indicate that there is a new feature in the theme editor page and add a `style` prop with a `zIndex` value of 1010 to ensure that the float button is above the header and the content ([link](https://github.com/ant-design/ant-design/pull/44111/files?diff=unified&w=0#diff-8d8f58ae995a6180f1788f819583cf51b719b8d8ab1ae422b06a1e8d846f2447L27-R33))
*  Remove the redundant `badge` prop from the `Link` component in the `ThemeSwitch` component ([link](https://github.com/ant-design/ant-design/pull/44111/files?diff=unified&w=0#diff-8d8f58ae995a6180f1788f819583cf51b719b8d8ab1ae422b06a1e8d846f2447R85))
*  Add a new property `contentMarginTop` to the interface `NewToken` and the `SiteThemeProvider` component to define the margin-top value for the main content area in the site theme and assign a default value of 40 ([link](https://github.com/ant-design/ant-design/pull/44111/files?diff=unified&w=0#diff-3bb66880942ca129db42a1a6c20126916fc7b9be5daf9723506df03fa96f0995R18), [link](https://github.com/ant-design/ant-design/pull/44111/files?diff=unified&w=0#diff-3bb66880942ca129db42a1a6c20126916fc7b9be5daf9723506df03fa96f0995R56))
*  Modify the `useStyle` hook in the `SidebarLayout`, `Content`, `Header`, and `Sidebar` components to use the theme token values of `contentMarginTop`, `headerHeight`, `marginXXL`, `mobileMaxWidth`, `colorBgContainer`, `boxShadowTertiary`, and `zIndexPopupBase` from the `token` object and replace the hard-coded values ([link](https://github.com/ant-design/ant-design/pull/44111/files?diff=unified&w=0#diff-0203e852c7cd27a2f6c67aea218485e507f4e9f18477d08de2fd0e19f0281f5eL8-R11), [link](https://github.com/ant-design/ant-design/pull/44111/files?diff=unified&w=0#diff-1533ad458dddb5cacbf24062cc5d262ca684a57e2754bf4777f24f37411d5f82L58-R62), [link](https://github.com/ant-design/ant-design/pull/44111/files?diff=unified&w=0#diff-2b1224dbeb7d20f1cb1b4590eb12b68104c605d2a67f0b0bd4fd9d8b0100a728L28-R34), [link](https://github.com/ant-design/ant-design/pull/44111/files?diff=unified&w=0#diff-df557d0ed53eb1710843e7fce2d294362941ec80e20a8bb0ed385299362e204bL108-R111))
*  Add a `zIndex` value of 1000 and a `backdropFilter` value of `blur(8px)` to the `header` style in the `useStyle` hook of the `Header` component to make the header stick to the top of the viewport and have a blurred background when scrolling ([link](https://github.com/ant-design/ant-design/pull/44111/files?diff=unified&w=0#diff-2b1224dbeb7d20f1cb1b4590eb12b68104c605d2a67f0b0bd4fd9d8b0100a728L28-R34))
*  Add a `zIndex` value of 1000 to the `tocWrapper` style in the `useStyle` hook of the `Content` component to ensure that the table of contents is above the header and the content ([link](https://github.com/ant-design/ant-design/pull/44111/files?diff=unified&w=0#diff-1533ad458dddb5cacbf24062cc5d262ca684a57e2754bf4777f24f37411d5f82R68))
*  Modify the `targetOffset` prop of the `Anchor` component in the `Content` component to use the `marginXXL` value from the theme token to ensure that the anchor links are aligned with the headings in the content ([link](https://github.com/ant-design/ant-design/pull/44111/files?diff=unified&w=0#diff-1533ad458dddb5cacbf24062cc5d262ca684a57e2754bf4777f24f37411d5f82L209-R233))
*  Remove the unused `Affix` and `Divider` components from the import statement in the `Content` component ([link](https://github.com/ant-design/ant-design/pull/44111/files?diff=unified&w=0#diff-1533ad458dddb5cacbf24062cc5d262ca684a57e2754bf4777f24f37411d5f82L9-R9))
*  Remove the redundant hook `useTheme` from the `IconSearch` component ([link](https://github.com/ant-design/ant-design/pull/44111/files?diff=unified&w=0#diff-8b799f7fc1319c44839ab462e8211e9e598d446618f5770aac912faebc536077L114))
*  Add a new function `withPureRenderTheme` to the `PurePanel` component in the `_util` folder to wrap a component with a `ConfigProvider` component that sets the theme token values of `motion` and `zIndexPopupBase` to false and 0 respectively ([link](https://github.com/ant-design/ant-design/pull/44111/files?diff=unified&w=0#diff-458739bbc5dc90c9430ef8ee4ea53366ddef7c5870bdaeb30ae45240c07777c6R5-R21))
*  Modify the `PurePanel` function in the `PurePanel` component in the `_util` folder to remove the type assertion, add a return statement, and change it from an arrow function to a regular function ([link](https://github.com/ant-design/ant-design/pull/44111/files?diff=unified&w=0#diff-458739bbc5dc90c9430ef8ee4ea53366ddef7c5870bdaeb30ae45240c07777c6L19-R36))
*  Add a `zIndexPopupBase` property to the `theme` prop of the `ConfigProvider` component in the `PurePanel` function in the `_util` folder and wrap the `PurePanel` function with the `withPureRenderTheme` function ([link](https://github.com/ant-design/ant-design/pull/44111/files?diff=unified&w=0#diff-458739bbc5dc90c9430ef8ee4ea53366ddef7c5870bdaeb30ae45240c07777c6R99), [link](https://github.com/ant-design/ant-design/pull/44111/files?diff=unified&w=0#diff-458739bbc5dc90c9430ef8ee4ea53366ddef7c5870bdaeb30ae45240c07777c6L97-R117))
*  Import the `withPureRenderTheme` function from the `PurePanel` component in the `_util` folder and wrap the `PurePanel` component with it in the `modal` and `tour` folders ([link](https://github.com/ant-design/ant-design/pull/44111/files?diff=unified&w=0#diff-9103d37c6668043c170813d54cb43ca28bc6cf111eedfc6718aeae7f540a0b27R12), [link](https://github.com/ant-design/ant-design/pull/44111/files?diff=unified&w=0#diff-9103d37c6668043c170813d54cb43ca28bc6cf111eedfc6718aeae7f540a0b27L83-R84), [link](https://github.com/ant-design/ant-design/pull/44111/files?diff=unified&w=0#diff-7348f4a7ff6e65656b442037c43f19497ece5b3d5031b19e39b0ead4c6016e07R8), [link](https://github.com/ant-design/ant-design/pull/44111/files?diff=unified&w=0#diff-7348f4a7ff6e65656b442037c43f19497ece5b3d5031b19e39b0ead4c6016e07L39-R40))

### CODE REVIEW

File: components/app/index.en-US.md [hunk 1]
```diff
@@ -6,6 +6,7 @@ cover: https://mdn.alipayobjects.com/huamei_7uahnr/afts/img/A*HJz8SZos2wgAAAAAAA
 coverDark: https://mdn.alipayobjects.com/huamei_7uahnr/afts/img/A*oC92TK44Ex8AAAAAAAAAAAAADrJ8AQ/original
 demo:
   cols: 2
+tag: New
```
Comment (kiner-tang) [inline components/app/index.en-US.md:9]: 原来我们的 New 标签是直接写在文档里面的吗？那什么时候要移除 New 呢？

Comment (MadCcc) [reply components/app/index.en-US.md:9]: 下个 major

**Review by kiner-tang — APPROVED** [review]

File: components/app/index.en-US.md (hunk 1 above)
Comment (MadCcc) [reply components/app/index.en-US.md:9]: 可以考虑从 changelog 里抓一下新组件的关键字，等下搞下

Comment (kiner-tang) [reply components/app/index.en-US.md:9]: 明白嘞

**Pull request merged**